internal/pkg/probe/probe.go

```diff
@@ -36,32 +36,40 @@ func New(b block.Interface, f *Filter) Interface {
 }
 
 // ScanAvailableDevices retrieves devices that are unformatted.
-func (m *deviceScanner) ScanAvailableDevices(ctx context.Context, log logr.Logger) (*block.DeviceList, error) {
+func (m *deviceScanner) ScanAvailableDevices(ctx context.Context) (*block.DeviceList, error) {
+	log := log.FromContext(ctx)
 	devices, err := m.GetDevices(ctx)
 	if err != nil {
 		return nil, fmt.Errorf("failed to get devices: %w", err)
 	}
 
-	var unformatted []block.Device
+	var availableDevices []block.Device
 	for _, device := range devices.Devices {
 		if !m.filter.Match(device) {
 			log.V(3).Info("device filtered out", "device", device)
 			continue
 		}
-		isFormatted, err := m.IsFormatted(device.Path)
+		fsType, isFormatted, err := m.IsFormatted(device.Path)
```

Comment:
I think I'd create a new `IsLVM2(path string) (bool, err)` to hide the fsType
check. Then `IsFormatted()` doesn't need to return the fstype. Both can be
wrappers for a shared func in block.go.